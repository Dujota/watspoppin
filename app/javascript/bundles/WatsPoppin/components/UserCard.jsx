import React from 'react';

export default class UserCard extends React.Component {
  render() {
    return (
      <figure>
        <img
          src="https://assets.vogue.com/photos/5891c8b097a3db337a24c8f7/master/pass/llamas-and-haircuts-prince-harry1.jpg"
          alt="temporary placeholder"
        />
        <figcaption>You are here</figcaption>
      </figure>
    );
  }
}
