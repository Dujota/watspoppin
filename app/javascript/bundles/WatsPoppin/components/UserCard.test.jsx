import UserCard from './UserCard';

// Test when there's a user
it('renders user card when user is present', () => {
  const user = {
    image: {
      thumb: { url: '/luigi.jpg' }
    },
    name: 'Maria Antonieta de las Nieves'
  };

  const wrapper = render(<UserCard user={user} />);
  expect(wrapper).toMatchSnapshot();
});

// Test when there's no user
it('renders user card when user is null', () => {
  const user = null;

  const wrapper = render(<UserCard user={user} />);
  expect(wrapper).toMatchSnapshot();
});
