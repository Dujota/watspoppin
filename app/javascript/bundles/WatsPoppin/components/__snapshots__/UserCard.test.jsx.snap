// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders user card when user is null 1`] = `
<figure>
  <img
    alt="super mario"
    src="/mario.jpg"
  />
  <figcaption>
    You're here!!
  </figcaption>
</figure>
`;

exports[`renders user card when user is present 1`] = `
<figure>
  <img
    alt="Maria Antonieta de las Nieves"
    src="/luigi.jpg"
  />
  <span
    class="Linkify"
  >
    <figcaption>
      Maria Antonieta de las Nieves
    </figcaption>
  </span>
</figure>
`;
